### ISSUE TITLE
REFRESH cannot run inside a transaction block

### ISSUE THREAD

**Issue by dvdrozdov** [timescale/timescaledb#1218]
**Relevant system information:**
 - OS: Ubuntu 18.04
 - PostgreSQL version: 10.7
 - TimescaleDB version: 1.3.0
 - Installation method: "apt install"

**Describe the bug**
After timescaledb 1.3.0 installation refreshing of a materialized view inside a transaction fails with the following error:
```
REFRESH cannot run inside a transaction block
```

**To Reproduce**
```
createdb -U postgres test

test=# create materialized view mvtest as select 1;
SELECT 1

test=# select * from mvtest;
 ?column? 
----------
        1
(1 row)

test=# refresh materialized view mvtest;
REFRESH MATERIALIZED VIEW
test=# begin;
BEGIN
test=# refresh materialized view mvtest;
REFRESH MATERIALIZED VIEW
test=# commit;
COMMIT

CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;
...
Running version 1.3.0

test=# begin;
BEGIN
test=# refresh materialized view mvtest;
ERROR:  REFRESH cannot run inside a transaction block
test=# commit;
ROLLBACK

test=# select version();
                                                             version                                                              
----------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 10.7 (Ubuntu 10.7-1.pgdg18.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0, 64-bit
(1 row)

dpkg -l | grep timesc
ii  timescaledb-loader-postgresql-10       1.3.0~ubuntu18.04                  amd64        The loader for TimescaleDB to load individual versions.
ii  timescaledb-postgresql-10              1.3.0~ubuntu18.04                  amd64        An open-source time-series database based on PostgreSQL, as an extension.
ii  timescaledb-tools                      0.6.0~ubuntu18.04                  amd64        A suite of tools that can be used with TimescaleDB.
```

**Comment by od0** [comment]
seeing this as well

**Comment by cevian** [comment]
Solved by #1236. Will be part of 1.3.1

**Comment by vi-vsn** [comment]
Hello,

I have originally ran into the issue reported in this thread while running version 1.3.0 of TimescaleDb.  This happened a few days ago. Today I have upgraded to 1.4.0 believing that this will resolve the problem, however it is now crashing in the exactly same routine, but with a different error message:

1.3.0: 

REFRESH cannot run inside a transaction block

1.4.0: 

FATAL: the database system is in recovery mode

Digging into the pglog give a little more context:

1. statement: select <redacted>_monthly_report()
2. server process (PID 3284) was terminated by signal 11: Segmentation fault
3. DETAIL:  Failed process was running: select <redacted>_monthly_report()
4. LOG:  terminating any other active server processes
5. WARNING:  terminating connection because of crash of another server process
6. DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
7. HINT:  In a moment you should be able to reconnect to the database and repeat your command.
8. LOG:  connection received
9. FATAL:  the database system is in recovery mode

The function in question performs the following tasks:

1. Checks whether a table already has the monthly_report records. The table is *not* a hypertable.
2. If monthly_report records exist, it returns them.
3. If monthly_report records don't exists - it queries various hyper_tables to produce them and writes the them into the monthly_report table to cache them for any subsequent calls and then returns them. It never writes into any hypertables, only reads.
